tests/integration/upsertGraph.js

```diff
@@ -1777,6 +1778,110 @@ module.exports = session => {
           });
       });
     });
+    describe('upserts with update: true option', () => {
+      before(() => {
+        Model1.$$jsonSchema = {
+          type: 'object',
+          required: ['model1Prop1'],
+
+          properties: {
+            model1Prop1: {type: 'string'}
+          }
+        };
+      });
+
+      after(() => {
+        delete Model1.$$jsonSchema;
+        delete Model1.$$validator;
+      });
+
+      it('should fail to do an incomplete upsert', () => {
+        const fails = [
+          {
+            id: 2,
+            // This fails because of missing property.
+            // model1Prop1: 'updated root 2',
+
+            model1Relation1: {
+              id: 3,
+              model1Prop1: 'updated belongsToOne'
+            }
+          },
+          {
+            id: 2,
+            model1Prop1: 'updated root 2',
+
+            model1Relation1: {
+              id: 3
+              // This fails because of missing property.
+              // model1Prop1: 'updated belongsToOne'
+            }
+          }
+        ];
+
+        const success = [
+          {
+            id: 2,
+            model1Prop1: 'updated root 2',
+
+            model1Relation1: {
+              id: 3,
+              model1Prop1: 'updated belongsToOne'
+            }
+          }
+        ];
+
+        return Promise.map(fails, fail => {
+          return transaction(session.knex, trx =>
+            Model1.query(trx).upsertGraph(fail, {update: true})
+          ).reflect();
+        })
```

Comment:
Sigh... I hate the way prettier aligns code with these. Well, there nothing we can do..

Reply:
I guess you can configure it?

Reply:
Nope, the there is not much to configure in prettier.

Reply:
That's why I use a heavily configured ESLint with precommit hooks instead of prettier. Prettier is great, but for me there are too many edge cases where it behaves a tad too weirdly.